src/librustc_metadata/rmeta/encoder.rs

```diff
@@ -228,8 +228,13 @@ impl<'tcx> SpecializedEncoder<LocalDefId> for EncodeContext<'tcx> {
     }
 }
 
-impl<'tcx> SpecializedEncoder<Ty<'tcx>> for EncodeContext<'tcx> {
-    fn specialized_encode(&mut self, ty: &Ty<'tcx>) -> Result<(), Self::Error> {
+impl<'a, 'b, 'tcx> SpecializedEncoder<&'a ty::TyS<'b>> for EncodeContext<'tcx>
```

Comment:
are the two separate lifetimes `'a` and `'b` really necessary here? Couldn't they just be one and use `Ty<'a>`?

Reply:
Unfortunately not. Specializing impls can't repeat type/lifetime parameters.